src/api/layers/layers-external.api.js

```diff
@@ -155,3 +157,29 @@ export function parseWmtsCapabilities(content, originUrl) {
         )
     }
 }
+
+const ENC_PIPE = '%7C'
+
+/**
+ * Encode an external layer parameter.
+ *
+ * This percent encode the special character | used to separate external layer parameters.
+ *
+ * @param {string} param Parameter to encode
+ * @returns {string} Percent encoded parameter
+ */
+export function encodeExternalLayerParam(param) {
+    return param.replace('|', ENC_PIPE)
+}
+
+/**
+ * Decode an external layer parameter.
+ *
+ * This percent decode the special character | used to separate external layer parameters.
+ *
+ * @param {string} param Parameter to encode
+ * @returns {string} Percent encoded parameter
+ */
+export function decodeExternalLayerParam(param) {
+    return param.replace(ENC_PIPE, '|')
+}
```

Comment:
I don't know if it wouldn't make more sense to use something like [encodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) and [decodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent) here instead...

Reply:
@pakb now otherwise it would also encode other characters that would be encoded twice, because the vue router is using `encodeURIComponent` so that would encode the caractere twice